checkstyle/suppressions.xml

```diff
@@ -85,6 +85,9 @@
     <suppress checks="NPathComplexity"
               files="SslTransportLayer.java"/>
 
+    <suppress checks="ImportControl"
+              files="KafkaFuture.java|KafkaFutureTest.java"/>
```

Comment:
We should just allow `common` to access `common.internals`, there's no good reason not to allow that. It's a simple addition:

<allow pkg="org.apache.kafka.common.internals" exact-match="true" />

In general, `import-control.xml` is the right place to make these changes, I don't think we should be using suppressions for this.